CoreFoundation/String.subproj/CFString.c

```diff
@@ -1264,12 +1264,15 @@ CF_PRIVATE CFStringRef __CFStringCreateImmutableFunnel3(
 	contentsDeallocator = __CFGetDefaultAllocator();
     }
 
+    #ifndef DEPLOYMENT_TARGET_ANDROID
+    // crashes String.swift, line 51 when creating String from zero length Data
```

Comment:
IIRC android's memcpy and memmove are not the same as Darwin. That may be the culprit of this issue.